### ISSUE TITLE
CloundSQL unexpectedly goes down from time to time

### ISSUE THREAD

**Issue by VasulBera** [GoogleCloudPlatform/cloud-sql-jdbc-socket-factory#125]
Faced with cloudSQL issue, from time to time I got my AppEngine not working, that caused a series of strange errors, seems like a problem in MySQL connector/cloud socket factory.
I can reproduce it in the next way: run an app on AppEngine, then reboot DB server, in the next about 10 minutes the app will work, then it goes down on about 20-30 minutes with the errors.
Errors that we get (No matter what query we send): 
- java.sql.SQLException: ResultSet is from UPDATE. No Data. 
- java.sql.SQLException: Column 'user_id1_12_' not found (obviously we do not have such a column)
- org.eclipse.jetty.io.EofException: Closed
- java.sql.SQLException: Unknown charater set index for field '507' received from server (obviously we do not have such a field)

My stack of technologies: 
- Spring boot 2.0.4
- Hibernate 5.2.17
- MySQL connector 5.1.47
- Mysql socket Factory 1.0.12
- AppEngine Standart
- MySQL 5.7

P.S. I tried to reproduce this issue connected directly to MySQL on AppEngine - it works good, problem occured when using CloudSQL proxy.

**Comment by kurtisvg** [comment]
So I'm going to start off an make an assumption that this is using the Java 8 App Engine Standard runtime (please correct me if I'm wrong). I'm a little unclear on what you suspect the issue is. 

You mentioned using the Cloud SQL proxy - are you talking about this SocketFactory? I'm not really sure how you would be running the Cloud SQL proxy on App Engine.

You also mentioned you connected directly to MySQL from App Engine - how? 

It looks like you are using an older (5.x) version of the MySQL connector - is there a particular reason you are pinned to this version? I believe the recommendation from the MySQL folks is to move up to 8.x (they are still compatible with MySQL 5.x).

**Comment by VasulBera** [comment]
I'm not sure is the issue on SocketFactory or in CloudSQL, so the reason why I wrote here - it could be related to SocketFactory. 

How did I connect to MySQL directly? - I Added all ip addresses to white list, and connect to the DB by URL which include IP and port (usual url to DB).

There is no any reason why I am using 5.x version of MySQL connector, I saw here (https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/master/examples/compute-engine/pom.xml) we have version 5.x and took it like an example.

**Comment by kurtisvg** [comment]
I would suggest updating `mysql-connector-java` to 8.x and switch to the matching version for the SocketFactory, and let us know if the issue persists. 

Are you Hikari for your connection pool? What settings are you using?

**Comment by scottlawrencelawson** [comment]
I am just following because this affects an app of ours. I am not a developer, but googling, I assume by "Are you Hikari for your connection pool?" you are asking if VasulBera is using this: https://github.com/brettwooldridge/HikariCP ?

**Comment by kurtisvg** [comment]
Yes, sorry if it was unclear. it was meant to be "Are you using Hikari...", just a typo on my part.

**Comment by VasulBera** [comment]
Hi everyone,

We have made a few tests yesterday and today, referring to kurtisvg note we deployed two new versions of the app with different connectors.
First was with MySQL connector 8.x and socket factory - mysql-socket-factory-connector-j-8:1.0.12
And Second with MySQL connector 6.x and socket factory - mysql-socket-factory-connector-j-6:1.0.12

Also we have tried the third scenario by deploying the version we used initially: 
MySQL connector 5.x and socket factory - mysql-socket-factory:1.0.12

Then we restarted DB server and tried to reproduce the issue. We repeated this tests three times and couldn't reproduce it.
Each connector (5.x, 6.x, 8.x) works good.

We are not sure who fixed this but at this point issue is resolved.
